### PR TITLE
internal/dag: Gateway Listeners with invalid references should have reason RefNotPermitted

### PR DESCRIPTION
**Pull request by sunjayBhatia**
See https://github.com/kubernetes-sigs/gateway-api/pull/1305

Fixes conformance test failures on main

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
this is one of those changes where we have a conflict between `main` of Contour + the latest GW API conformance release

**Comment by skriss** [comment]
> this is one of those changes where we have a conflict between `main` of Contour + the latest GW API conformance release

Yeah, I guess this brings up a new issue here - if we merge this, then `main` Gateway conformance will pass, but PR Gateway conformance will start failing since this is not compatible with the v0.5.0 conformance tests.

Perhaps it makes sense to hold this PR open until the upstream conformance change is included in a released version (hopefully v0.5.1)?

**Comment by sunjayBhatia** [comment]
> > this is one of those changes where we have a conflict between `main` of Contour + the latest GW API conformance release
> 
> Yeah, I guess this brings up a new issue here - if we merge this, then `main` Gateway conformance will pass, but PR Gateway conformance will start failing since this is not compatible with the v0.5.0 conformance tests.
> 
> Perhaps it makes sense to hold this PR open until the upstream conformance change is included in a released version (hopefully v0.5.1)?

yeah, was thinking that would probably have to be the case

submitted: https://github.com/kubernetes-sigs/gateway-api/issues/1328 to start the conversation

File: internal/dag/gatewayapi_processor.go [hunk 1]
```diff
@@ -695,7 +695,7 @@ func (p *GatewayAPIProcessor) resolveListenerSecret(certificateRefs []gatewayapi
 			listenerName,
 			gatewayapi_v1beta1.ListenerConditionResolvedRefs,
 			metav1.ConditionFalse,
-			gatewayapi_v1beta1.ListenerReasonInvalidCertificateRef,
+			gatewayapi_v1beta1.ListenerReasonRefNotPermitted,
```
Comment (skriss) [inline internal/dag/gatewayapi_processor.go:698]: Based on https://github.com/kubernetes-sigs/gateway-api/pull/1334 it looks to me like this one should have its reason stay as is -- i.e. `InvalidCertificateRef`.

**Review by skriss — APPROVED** [review]
LGTM

**Pull request merged**